app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/service.tsx

```diff
@@ -10,20 +10,107 @@ import * as loggerProvider from "../providers/logger";
 // === Constants ===
 // =================
 
+/** Default HTTP body for an "open project" request. */
+const DEFAULT_OPEN_PROJECT_BODY: OpenProjectRequestBody = {
+  forceCreate: false,
+};
+
 /** Relative HTTP path to the "get user" endpoint of the Cloud backend API. */
 const GET_USER_PATH = "users/me";
+/** Relative HTTP path to the "list projects" endpoint of the Cloud backend API. */
+const LIST_PROJECTS_PATH = "projects";
+/** Relative HTTP path to the "create project" endpoint of the Cloud backend API. */
+const CREATE_PROJECT_PATH = "projects";
+/** Relative HTTP path to the "close project" endpoint of the Cloud backend API. */
+const closeProjectPath = (projectId: ProjectId) =>
+  `projects/${projectId}/close`;
+/** Relative HTTP path to the "get project" endpoint of the Cloud backend API. */
+const getProjectPath = (projectId: ProjectId) => `projects/${projectId}`;
+/** Relative HTTP path to the "open project" endpoint of the Cloud backend API. */
+const openProjectPath = (projectId: ProjectId) => `projects/${projectId}/open`;
 
 // =============
 // === Types ===
 // =============
 
+/** Unique identifier for a user's project. */
+export type ProjectId = string;
```

Comment:
use brand here?

Reply:
actually i've implemented all endpoints that should be accessible from the frontend [here](https://github.com/enso-org/enso/blob/5fb66bb0679db2a62b8f43398e565f2e2e708ba9/app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/service.tsx)

([here](https://github.com/enso-org/enso/blob/8a3f1836912101e71e7cf870e19dcb97ce90a53d/app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/service.tsx#L1) is the same file but with the new endpoints+parameters for the new directory-based access. it hasn't been tested yet though)

maybe `dashboard/service.tsx` should be extracted to its own PR instead?

Reply:
Yes, it will be extracted to its own PR.